src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -131,11 +131,25 @@ protected override void CheckInterfaces(DiagnosticBag diagnostics)
                 var corLibrary = this.ContainingAssembly.CorLibrary;
                 var conversions = new TypeConversions(corLibrary);
                 var location = singleDeclaration.NameLocation;
+                var seenInterfaces = PooledHashSet<NamedTypeSymbol>.GetInstance();
 
                 foreach (var @interface in interfaces)
                 {
                     @interface.CheckAllConstraints(conversions, location, diagnostics);
+
+                    foreach (var seenInterface in seenInterfaces)
+                    {
+                        if (@interface.Equals(seenInterface, TypeCompareKind.IgnoreTupleNames))
+                        {
+                            diagnostics.Add(ErrorCode.ERR_DuplicateInterfaceWithTupleNamesInBaseList, location, @interface, this);
```

Comment:
may be nice to report seenInterface  as well. It could be a very puzzling error, so more information is better.